pkgs/development/libraries/science/math/p4est-sc/default.nix

```diff
@@ -0,0 +1,67 @@
+{ lib, stdenv, fetchFromGitHub
+, autoreconfHook, pkg-config
+, p4est-sc-debugEnable ? true, p4est-sc-mpiSupport ? true
+, mpi ? null, openmpi ? null, openssh ? null, zlib
+}:
+
+# we prefer MPICH over OpenMPI; call accordingly
+assert p4est-sc-mpiSupport -> mpi != null;
+assert p4est-sc-mpiSupport && mpi == openmpi -> openssh != null;
+
+let
+  dbg = if debugEnable then "-dbg" else "";
+  debugEnable = p4est-sc-debugEnable;
+  mpiSupport = p4est-sc-mpiSupport;
+in
+stdenv.mkDerivation {
+  pname = "p4est-sc${dbg}";
+  version = "unstable-2021-06-14";
+
+  # fetch an untagged snapshot of the prev3-develop branch
+  src = fetchFromGitHub {
+    # url = "https://github.com/cburstedde/libsc.git";
+    owner = "cburstedde";
+    repo = "libsc";
+    rev = "1ae814e3fb1cc5456652e0d77550386842cb9bfb";
+    sha256 = "14vm0b162jh8399pgpsikbwq4z5lkrw9vfzy3drqykw09n6nc53z";
+  };
+
+  nativeBuildInputs = [ autoreconfHook pkg-config ];
+  propagatedBuildInputs = [ zlib ]
+    ++ lib.optional mpiSupport mpi
+    ++ lib.optional (mpiSupport && mpi == openmpi) openssh
+  ;
+  inherit debugEnable mpiSupport;
+
+  postPatch = ''
+    echo "dist_scaclocal_DATA += config/sc_v4l2.m4" >> Makefile.am
+  '';
+  preConfigure = ''
+    echo "2.8.0" > .tarball-version
+    ${if mpiSupport then "unset CC" else ""}
+  '';
+
+  configureFlags = [ ]
+    ++ lib.optional debugEnable "--enable-debug"
```

Comment:
```suggestion
  configureFlags = lib.optional debugEnable "--enable-debug"
```

Reply:
Thanks for your hints so far. The packages look a lot cleaner now.

The above is for symmetry between p4est:

    configureFlags = [ "--with-sc=${p4est-sc}" ]
      ++ lib.optional debugEnable "--enable-debug"
      ++ lib.optional mpiSupport "--enable-mpi"
    ; 

and p4est-sc:

    configureFlags = [ ]
      ++ lib.optional debugEnable "--enable-debug"
      ++ lib.optional mpiSupport "--enable-mpi"
    ;

Just wondering whether you could let that slide as a matter of style. Alternatively, I might offer:

    configureFlags =
         lib.optional debugEnable "--enable-debug"
      ++ lib.optional mpiSupport "--enable-mpi"
    ;

Reply:
lib.optional returns an empty array when false. You might want to consider using a common nix file and build the two packages from them. I think php and pythonPackages.pillow are two examples doing it like that.

Reply:
Thanks, I thought about that a bit. However, there are several similar upcoming packages using p4est-sc, such as t8code and forestclaw, and I'd like to keep p4est-sc separate as a common point of reference. I also thought about inheriting p4est-sc.configureFlags into p4est, but i might do that some time later. Right now I can just squash and be done with it.